api.js

```diff
@@ -247,17 +248,29 @@ class Api extends EventEmitter {
 			});
 		});
 
+		let interrupted = false;
+
 		return this._computeForkExecArgs(files)
 			.then(argvList => {
 				execArgvList = argvList;
 			})
 			.return(files)
 			.map((file, index) => {
-				return new Promise(resolve => {
+				return new Promise((resolve, reject) => {
 					const forkArgs = execArgvList[index];
+					if (interrupted) {
+						const relFile = path.relative('.', file);
+						return reject(new AvaError(`File ${relFile} was not run.`));
+					}
+
 					const test = this._runFile(file, runStatus, forkArgs);
 					tests.push(test);
 
+					process.on('SIGINT', () => {
```

Comment:
This will trigger "memory leak" warnings when there are more than 10 test files. Each `test` is pushed onto the `tests` array though so you could use that to set up one `SIGINT` listener and exit each active test.